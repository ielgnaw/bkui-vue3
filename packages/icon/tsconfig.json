{
  "ts-node": {
    "transpileOnly": true
  },
  "compilerOptions": {
    "target": "ES2019",
    "module": "commonjs",
    "strict": true,
    "declaration": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "jsx": "preserve",
    "jsxFactory": "h"
  },
  "include": [
    "./src/**/*",
    "./scripts/*/*"
  ],
  "exclude": [
    "*.js"
  ]
}