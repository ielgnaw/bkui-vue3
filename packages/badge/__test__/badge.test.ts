/*
 * Tencent is pleased to support the open source community by making
 * 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community Edition) available.
 *
 * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
 *
 * 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community Edition) is licensed under the MIT License.
 *
 * License for 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community Edition):
 *
 * ---------------------------------------------------
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
 * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
 * the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
 * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
 * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */

import Button from '@bkui-vue/button';
import { Help } from '@bkui-vue/icon';
import { mount } from '@vue/test-utils';

import Badge from '../src/index';
const mountTpl = (template: string) =>
  mount(
    {
      components: {
        bkBadge: Badge,
        Help,
        bkButton: Button,
      },
      template,
    },
    {
      global: {
        provide: {
          badge: {},
        },
      },
    },
  );
describe('Badge.tsx', () => {
  it('theme', async () => {
    const theme = 'danger';
    const wrapper = await mount(Badge, {
      props: { theme },
    });
    expect(wrapper.findAll(`.bk-${theme}`).length).toEqual(1);
  });

  it('count', async () => {
    const wrapper = await mount(Badge, {
      props: { count: 10 },
    });
    // todo
    expect(wrapper.findAll('.bk-primary').length).toEqual(1);
  });
  it('dot', async () => {
    const wrapper = await mount(Badge, {
      props: { dot: true },
    });
    expect(wrapper.findAll('.dot').length).toEqual(1);
  });
  it('overflowCount', async () => {
    const wrapper = await mount(Badge, {
      props: { count: 100 },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('position', async () => {
    const wrapper = await mount(Badge, {
      props: { position: 'top-left' },
    });
    expect(wrapper.findAll('.top-left').length).toEqual(1);
  });
  it('slot', async () => {
    const wrapper = mountTpl(`
    <bkBadge>
      <bkButton>Badge</bkButton>
    </bkBadge>`);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('icon', async () => {
    const wrapper = mountTpl(`
    <bkBadge>
      <template slot="icon">
        <Help />
      </template>
      <bkButton>Badge</bkButton>
    </bkBadge>`);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('visible', async () => {
    const wrapper = await mount(Badge, {
      props: { visible: true },
    });
    expect(wrapper.findAll('.bk-badge').length).toEqual(0);
  });
});
