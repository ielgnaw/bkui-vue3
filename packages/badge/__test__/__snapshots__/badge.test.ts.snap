// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge.tsx icon 1`] = `
"<div class=\\"bk-badge-main\\"><template slot=\\"icon\\"></template><button class=\\"bk-button\\">
    <!----><span class=\\"bk-button-text\\">Badge</span>
  </button><span class=\\"bk-badge bk-primary pinned  top-right\\"><span>1</span></span></div>"
`;

exports[`Badge.tsx overflowCount 1`] = `"<div class=\\"bk-badge-main\\"><span class=\\"bk-badge bk-primary\\"><span>99+</span></span></div>"`;

exports[`Badge.tsx slot 1`] = `
"<div class=\\"bk-badge-main\\"><button class=\\"bk-button\\">
    <!----><span class=\\"bk-button-text\\">Badge</span>
  </button><span class=\\"bk-badge bk-primary pinned  top-right\\"><span>1</span></span></div>"
`;
